apache/tests/test_apache.py

```diff
@@ -0,0 +1,124 @@
+# (C) Datadog, Inc. 2010-2018
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import pytest
+import os
+import subprocess
+import requests
+import time
+import logging
+
+from datadog_checks.apache import Apache
+
+log = logging.getLogger('test_apache')
+
+CHECK_NAME = 'apache'
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+HOST = os.getenv('DOCKER_HOSTNAME', 'localhost')
+PORT = '18180'
+BASE_URL = "http://{0}:{1}".format(HOST, PORT)
+
+STATUS_URL = "{0}/server-status".format(BASE_URL)
+AUTO_STATUS_URL = "{0}?auto".format(STATUS_URL)
+
+CONFIG_STUBS = [
+    {
+        'apache_status_url': STATUS_URL,
+        'tags': ['instance:first']
+    },
+    {
+        'apache_status_url': AUTO_STATUS_URL,
+        'tags': ['instance:second']
+    },
+]
+BAD_CONFIG = {
+        'apache_status_url': 'http://localhost:1234/server-status',
+}
+
+APACHE_GAUGES = [
+    'apache.performance.idle_workers',
+    'apache.performance.busy_workers',
+    'apache.performance.cpu_load',
+    'apache.performance.uptime',
+    'apache.net.bytes',
+    'apache.net.hits',
+    'apache.conns_total',
+    'apache.conns_async_writing',
+    'apache.conns_async_keep_alive',
+    'apache.conns_async_closing'
+]
+
+APACHE_RATES = [
+    'apache.net.bytes_per_s',
+    'apache.net.request_per_s'
+]
+
+
+def wait_for_apache():
+    for _ in xrange(0, 100):
+        res = None
+        try:
+            res = requests.get(STATUS_URL)
+            res.raise_for_status
+            return
+        except Exception as e:
+            log.info("exception: {0} res: {1}".format(e, res))
+            time.sleep(2)
+    raise Exception("Cannot start up apache")
+
+
+@pytest.fixture(scope="session")
+def spin_up_apache():
+    env = os.environ
+    env['APACHE_CONFIG'] = os.path.join(HERE, 'compose', 'httpd.conf')
+    env['APACHE_DOCKERFILE'] = os.path.join(HERE, 'compose', 'Dockerfile')
+    args = [
+        "docker-compose",
+        "-f", os.path.join(HERE, 'compose', 'apache.yaml')
+    ]
+    subprocess.check_call(args + ["up", "-d", "--build"], env=env)
+    wait_for_apache()
+    for _ in xrange(0, 100):
+        requests.get(BASE_URL)
```

Comment:
What does this do?

Reply:
Seeds data into apache

Reply:
We're doing the same thing here: https://github.com/DataDog/integrations-core/pull/1314/files#diff-d9d1b38b7b7859b6a9278d0bffba94dbL30

Reply:
(This is a pretty frequent pattern in these tests where we do some requests or transactions to seed a bunch of data)